anndata/_core/anndata.py

```diff
@@ -664,10 +666,18 @@ def X(self, value: Optional[Union[np.ndarray, sparse.spmatrix]]):
         if (
             np.isscalar(value)
             or (hasattr(value, "shape") and (self.shape == value.shape))
+            or (
+                isinstance(value, AwkArray)
+                and (self.shape == (dim_len(value, 0), dim_len(value, 1)))
+            )
```

Comment:
Why do we actually need all these consistency checks here? Isn't the same thing checked by the `AlignedMapping` class?

Reply:
`X` is not stored in an `AlignedMapping`, so we would need to check the shape somewhere.